src/main/java/bio/terra/cli/service/utils/SamService.java

```diff
@@ -168,6 +169,25 @@ public void getOrRegisterUser() {
     }
   }
 
+  /**
+   * Call the SAM "/api/google/v1/user/proxyGroup/{email}" endpoint to get the email for the current
+   * user's proxy group.
+   *
+   * <p>Update the Terra User object with the proxy group email.
+   */
+  public void getProxyGroupEmail() {
```

Comment:
it's a little surprising to me that a `get` function returns void and mutates state. Couple ideas, a) rename to `storeProxyGroupEmail`, b) return the value and modify the terraUser in the calling function i.e. `AuthenticationManager`, or c) make TerraUser a smarter class that has `getOrRetrieveProxyGroupEmail` that grabs it from local storage or finds it in Sam as needed. I see this follows existing functions `getUser` and `getOrRegisterUser`, so also reasonable to change all in a later PR.

Reply:
Agreed. Updated them to return values instead.